.github/workflows/s3_tests.yml

```diff
@@ -0,0 +1,38 @@
+name: s3 fsspec tests
+
+on: pull_request
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        python-version: ["3.10"]
+        include:
+          - os: ubuntu-latest
+            python-version: 3.9
+          - os: ubuntu-latest
+            python-version: 3.8
```

Comment:
Do we need to have Python 3.8 and 3.9 tests for this? Probably Python 3.10 across the three OS is enough, but I'll leave that up to you

Reply:
Probably not but it's free testing... I think leave it for now but i don't feel that strongly. We might also want to add new tests for python 3.11-3.13 but we can leave that for future PRs.

Reply:
Well, it's definitely not free - there are definitely [usage limits on GitHub](https://docs.github.com/en/actions/administering-github-actions/usage-limits-billing-and-administration#usage-limits), but you're probably right that we wouldn't hit them unless we do extensive development on rex